### PR TITLE
Data export

### PR DESCRIPTION
**Pull request by Rand3rino**

### CODE REVIEW

**Comment by Rand3rino** [comment]
I created a separate project for the export similar to the import, which changed the .sln file. I deleted the project reference in the file but don' know how to resolve the conflict

**Comment by MarcusWichelmann** [comment]
> I created a separate project for the export similar to the import, which changed the .sln file. I deleted the project reference in the file but don' know how to resolve the conflict

I just resolved it somehow, don't think it will blow up. :smile:

File: Nesteo.Server/Controllers/Api/InspectionsController.cs [hunk 1]
```diff
@@ -164,5 +165,18 @@ public async Task<ActionResult<InspectionPreview>> GetInspectionPreviewByIdAsync
 
             return inspectionPreview;
         }
+
+        /// <summary>
+        /// Export all inspections
+        /// </summary>
+        [HttpPatch("export")]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public IAsyncEnumerable<string> ExportInspectionsAsync()
```
Comment (MarcusWichelmann) [inline Nesteo.Server/Controllers/Api/InspectionsController.cs]: `[HttpGet("csv")]` would be a better fit for this download-function I think.
Please change the name/summary to something in the sense of "Download CSV" then.
Status code is 200 OK.

Same for the nesting boxes please.

File: Nesteo.Server/Services/Implementations/InspectionService.cs [hunk 2]
```diff
@@ -45,6 +45,17 @@ public IAsyncEnumerable<InspectionPreview> GetAllPreviewsForNestingBoxIdAsync(st
                            .ProjectTo<InspectionPreview>(Mapper.ConfigurationProvider).AsAsyncEnumerable();
         }
 
+        public IAsyncEnumerable<string> ExportAllRowsAsync()
+        {
+            return Entities.AsNoTracking().OrderBy(entity => entity.Id).
+                            ProjectTo<InspectionExportRow>(Mapper.ConfigurationProvider).
+                            Select(row => String.Join(",", row.Id, row.NestingBoxId, row.InspectionDate, row.InspectionUser.UserName,
```
[the remark this answers is not in the document]
Comment (Rand3rino) [reply Nesteo.Server/Services/Implementations/InspectionService.cs]: Are you saying change to Select(string => String.Join(",", string.Id . . .)?

**Pull request merged**